src/System.Windows.Forms/src/System/Windows/Forms/MessageBox.cs

```diff
@@ -410,22 +387,24 @@ public static DialogResult Show(IWin32Window owner, string text)
             }
 
             Application.BeginModalMessageLoop();
-            DialogResult result;
             try
             {
-                result = Win32ToDialogResult(MessageBoxW(new HandleRef(owner, handle), text, caption, style));
+                ID mbresult = MessageBoxW(new HandleRef(owner, handle), text, caption, style);
+                return mbresult switch
+                {
+                    _ => (DialogResult)mbresult,
+                };
```

Comment:
In this case I think we can just do this:
```suggestion
                ID mbresult = MessageBoxW(new HandleRef(owner, handle), text, caption, style);
                return (DialogResult)mbresult;
```

Reply:
After applying it I think it could be reduced further to a single line within the ``try`` block:

```cs
return (DialogResult)MessageBoxW(new HandleRef(owner, handle), text, caption, style);
```